### PR TITLE
Empty all trashes on Linux

### LINKED ISSUE #9: doesn't empty trashes on other partitions
I have an ntfs partition mounted in ubuntu and this doesn't empty its trash.
The nautilus trash can do that (as it shows an aggregated view of all trashes from all disks).

The solution might be to list all mounted drives ([drivelist](https://github.com/resin-io/drivelist)), find the correct location of trash folders on these (e.g. `/mnt/drive123/.Trash-1000/`) and clean all of them.

If you're interested I could try to make a PR with this sometime.

### PR DESCRIPTION
**Pull request by tborychowski**
- find all trashes on linux (for current user)
- a bit of refactoring for readability
- tested on ubuntu 15.10

-

Fixes #9

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,11 +1,44 @@
 'use strict';
 const fs = require('fs');
+const os = require('os');
 const path = require('path');
 const childProcess = require('child_process');
 const runApplescript = require('run-applescript');
 const pify = require('pify');
 const rimraf = require('rimraf');
-const xdgTrashdir = require('xdg-trashdir');
```
Comment (sindresorhus) [inline index.js]: I think it would be better to make [`xdg-trashdir`](https://github.com/kevva/xdg-trashdir) support returning all the trash dirs instead of having all the logic here.

```js
xdgTrashdir.all();
```

File: index.js [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+const drivelist = require('drivelist');
```
Comment (sindresorhus) [inline index.js]: Use https://github.com/sindresorhus/df

File: index.js (hunk 1 above)
Comment (tborychowski) [inline index.js]: agreed. I will ask them

Comment (sindresorhus) [inline index.js]: I have access to that repo, so you could just do a pull request there and I'll review.

Comment (tborychowski) [inline index.js]: on second thought I think I'll pass on that :-) 
It seems that to properly extend that module I'd have to break the signature of the `export`...
(otherwise it would get ugly)

Comment (sindresorhus) [inline index.js]: > It seems that to properly extend that module I'd have to break the signature of the export...

No, you just do `module.exports.all = function () { ...`. I do that all the time in my modules to add extra functionality.

Some of the code there could probably be shared too, since it's doing almost the same thing, just for different drives.

Comment (sindresorhus) [inline index.js]: Looks like you could pretty much reuse most of this logic https://github.com/kevva/xdg-trashdir/blob/345387560a38dafb5aaddadba6cbfa7d2b3620a6/index.js#L31-L50

Comment (kevva) [inline index.js]: I'll add it.

**Comment by kevva** [comment]
Could you update this PR to reflect the changes from https://github.com/kevva/xdg-trashdir/commit/19678aa98d7431de308b9f01a0b5667e787dcdc5 @tborychowski?

**Comment by tborychowski** [comment]
@kevva, will do after work. 
Apparently I cannot install scoped packages in my corp:
```sh
npm ERR! unregistered users are not allowed to access package @sindresorhus/df : @sindresorhus/df
```

File: package.json [hunk 3]
```diff
@@ -37,11 +37,11 @@
     "pify": "^2.2.0",
     "rimraf": "^2.4.3",
     "run-applescript": "^2.0.0",
-    "xdg-trashdir": "^2.0.0"
+    "xdg-trashdir": "^2.1.0",
+    "path-exists": "^2.0.0"
```
Comment (kevva) [inline package.json]: Move this above `pify`. `npm install path-exists --save` adds it to `package.json` automatically in alphabetic order.

File: index.js [hunk 4]
```diff
@@ -16,18 +22,7 @@ module.exports = () => {
...
-		return Promise.all(paths.map(pth =>
-			pify(fs.readdir)(pth).then(files =>
-				Promise.all(files.map(file =>
-					pify(rimraf)(path.join(pth, file)))
-				)
-			)
-		));
-	});
+	return xdgTrashdir.all()
+		.then(dirs => dirs.filter(pathExists.sync))
```
Comment (kevva) [inline index.js]: `.then(dirs => Promise.all(dirs.filter(pathExists)))`

File: package.json [hunk 5]
```diff
@@ -37,11 +37,11 @@
...
   },
   "devDependencies": {
     "ava": "*",
-    "path-exists": "^2.0.0",
```
Comment (kevva) [inline package.json]: Keep it here too if it's needed in the tests. They might break in the future if we remove it from the `dependencies`.

File: index.js (hunk 4 above)
Comment (tborychowski) [inline index.js]: no, but `emptyTrash` function first finds all files in that location (it doesn't remove the `Trash` folder itself, only its contents), so `fs.readdir` on a non-existing folder would fail.

File: package.json (hunk 5 above)
Comment (tborychowski) [inline package.json]: following this logic, you'd need to list the same dependency 5 times if you use it in 5 places ;-)

If you remove a `dependency` that you use in prod. and you still need it in `dev` - you can then move it back to `devDependencies`. You should not list the same dep. more than once.

File: package.json (hunk 3 above)
Comment (tborychowski) [inline package.json]: I'm sure (or at least I hope) you have better things to do, than manually sorting lines in text files...
I know I have :-)

File: package.json (hunk 5 above)
Comment (kevva) [inline package.json]: True, but that never happens.

File: package.json (hunk 3 above)
Comment (SamVerschueren) [inline package.json]: @tborychowski It's all about perfectionism ;)

File: index.js (hunk 4 above)
Comment (kevva) [inline index.js]: Oh, yeah, then change it to `.then(dirs => Promise.all(dirs.filter(pathExists)))` :).

File: package.json (hunk 3 above)
Comment (sindresorhus) [inline package.json]: @tborychowski It's also because if we didn't, it would cause unrelated diff noise for the next commit that adds a dependency, because this dependency would be sorted by npm.

File: package.json (hunk 5 above)
Comment (sindresorhus) [inline package.json]: I agree. Dependencies should only be listed once.

**Comment by kevva** [comment]
@tborychowski, released in [2.1.0](https://github.com/sindresorhus/empty-trash/commit/3ddfcf50527e5ac4e72e8f920c44c91a98df142f)!
